source/LoadPanel.cpp

```diff
@@ -43,45 +43,44 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include <algorithm>
 #include <cstdlib>
-#include <iomanip>
-#include <sstream>
 #include <stdexcept>
 #include <utility>
 
 using namespace std;
 
 namespace {
 	// Return a pair containing settings to use for time formatting.
-	pair<const char*, const char*> TimestampFormatString(Preferences::DateFormat format)
+	pair<pair<string, string>, size_t> TimestampFormatString(Preferences::DateFormat fmt)
 	{
-		// pair<string, string>: Linux (1st) and Windows (2nd) format strings.
-		switch(format)
-		{
-			case Preferences::DateFormat::YMD:
-				return make_pair("%F %T", "%F %T");
-			case Preferences::DateFormat::MDY:
-				return make_pair("%-I:%M %p on %b %-d, %Y", "%#I:%M %p on %b %#d, %Y");
-			case Preferences::DateFormat::DMY:
-			default:
-				return make_pair("%-I:%M %p on %-d %b %Y", "%#I:%M %p on %#d %b %Y");
-		}
+		// pair<string, string>: Linux (1st) and Windows (2nd) format strings
+		// size_t: BUF_SIZE
+		if(fmt == Preferences::DateFormat::YMD)
+			return make_pair(make_pair("%F %T", "%F %T"), 26);
+		if(fmt == Preferences::DateFormat::MDY)
+			return make_pair(make_pair("%-I:%M %p on %b %-d, %Y", "%#I:%M %p on %b %#d, %Y"), 25);
+		if(fmt == Preferences::DateFormat::DMY)
+			return make_pair(make_pair("%-I:%M %p on %-d %b %Y", "%#I:%M %p on %#d %b %Y"), 24);
+
+		// Return YYYY-MM-DD by default.
+		return make_pair(make_pair("%F %T", "%F %T"), 26);
 	}
 
 	// Convert a time_t to a human-readable time and date.
 	string TimestampString(time_t timestamp)
 	{
-		pair<const char*, const char*> format = TimestampFormatString(Preferences::GetDateFormat());
-		stringstream ss;
+		pair<pair<string, string>, size_t> fmt = TimestampFormatString(Preferences::GetDateFormat());
+		char* buf = static_cast<char*>(std::malloc(fmt.second));
 
 #ifdef _WIN32
 		tm date;
 		localtime_s(&date, &timestamp);
-		ss << std::put_time(&date, format.second);
+		auto str = string(buf, strftime(buf, fmt.second, fmt.first.second.c_str(), &date));
```

Comment:
We shouldn't be completely reverting that commit, as the commit fixed an issue while introducing another. Reverting this commit just reintroduces the old issue.

The new issue is that strftime and std::put_time behave differently. If I rerun this code with the time format being  `%I:%M %p on %b %d, %Y` instead of `%#I:%M %p on %b %#d, %Y` then it works; i.e. I removed the two `#`s.

I'm wondering if the Windows vs Linux ifdef is unnecessary now and is handled by put_time.

Reply:
It looks like it behaves differently between clang and mingw, at least on Windows. The original report seems to be on Linux.

Reply:
The Linux problem I think is the hyphen, as the [documentation](https://cplusplus.com/reference/iomanip/put_time/) doesn’t show any hyphens.

Reply:
It looks like [microsoft put_time](https://learn.microsoft.com/en-us/cpp/standard-library/iomanip-functions?view=msvc-170#iomanip_put_time) takes a non constant time pointer and [unix put_time](https://en.cppreference.com/w/cpp/io/manip/put_time) takes a constant time pointer